src/main/java/sirius/web/security/UserInfo.java

```diff
@@ -167,7 +169,23 @@ public Builder withLang(String lang) {
          */
         public Builder withPermissions(Set<String> permissions) {
             verifyState();
-            user.permissions = permissions;
+            if (permissions == null) {
+                user.permissions = new HashSet<>();
+            } else {
+                user.permissions = permissions;
+            }
+            return this;
+        }
+
+        /**
+         * Indicates whether the user has all available permissions by default
+         *
+         * @param hasEveryPermission true when the user should have all permissions
+         * @return the builder itself for fluent method calls
+         */
+        public Builder withEveryPermission(boolean hasEveryPermission) {
```

Comment:
one could also make this "grantEveryPermission()" - as this is probably never invoked with false?

Reply:
I thought about this - I saw no benefit in terms of readability (`.withEveryPermission(true)` vs `.grantEveryPermission()`), it breaks the naming schema and it cannot be overridden later, so I kept this version. What benefits do you see from the signature change?